romancal/pipeline/exposure_pipeline.py

```diff
@@ -72,10 +77,53 @@ def process(self, input):
         if input_filename:
             result.meta.filename = input_filename
         result = self.saturation(result)
+
+        # Test for fully saturated data
+        if test_full_saturation(result):
+            log.debug('Error: all data saturated. When all of the pixels are saturated, the '
+                          'pipeline does not generate scientifically valid data. Accordingly, '
+                          'the pipeline is stopping and returning a zeroed-out image file.')
+
+            # Create zeroed out image file
+            # The set order is: data, dq, var_poisson, var_rnoise, err
+            result = ramp_fit_step.create_image_model(result,
+                                    (np.zeros(result.data.shape[1:], dtype=result.data.dtype),
+                                     result.pixeldq | result.groupdq[0] | dqflags.group['SATURATED'],
+                                     np.zeros(result.err.shape[1:], dtype=result.err.dtype),
+                                     np.zeros(result.err.shape[1:], dtype=result.err.dtype),
+                                     np.zeros(result.err.shape[1:], dtype=result.err.dtype)))
+
+            # Set all subsequent steps to skipped
+            for step_str in ['linearity', 'dark', 'jump', 'ramp_fit', 'assign_wcs',
+                             'flat_field', 'photom']:
+                result.meta.cal_step[step_str] = 'SKIPPED'
+
+            # Set suffix for proper output naming
+            self.suffix = 'cal'
+
+            return result
+
         result = self.linearity(result)
         result = self.dark_current(result)
         result = self.jump(result)
         result = self.rampfit(result)
+
+        # Test for fully saturated data
+        if "groupdq" in result.keys():
+            if test_full_saturation(result):
+                log.debug('Error: all data saturated. When all of the pixels are saturated, the '
+                          'pipeline does not generate scientifically valid data. Accordingly, '
+                          'the pipeline is stopping and returning a zeroed-out image file.')
```

Comment:
STCAL ramp_fit does return a None in this case, which gets turned into a zeroed-out cal image within RomanCAL RampFitStep, which is returned to the pipeline. (Otherwise _result_ gets overwritten with None, which means that we'd have to store a copy in the pipeline just for the ramp fit step.)